org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/fields/smartfield/AbstractSmartField.java

```diff
@@ -838,6 +865,9 @@ public void lookupByText(String text) {
   }
 
   protected void lookupByTextInternal(String text, boolean synchronous) {
+    if (StringUtility.length(text) > getMaxLength()) {
```

Comment:
I guess the If here is not required because the substring method deals with the case, that the string length <= max-length. So you could always call substring